src/raw/document.rs

```diff
@@ -633,7 +577,8 @@ impl ToOwned for RawDocument {
     type Owned = RawDocumentBuf;
 
     fn to_owned(&self) -> Self::Owned {
-        self.to_raw_document_buf()
+        // unwrap is ok here because we already verified the bytes in `RawDocumentRef::new`
```

Comment:
I think this copied comment is outdated - this should reference `RawDocument::decode_from_bytes`

This also got me thinking that `decode_from_bytes` read as a little confusing for these method names; "decode" implies to me that the bytes are being parsed, but we're actually just checking a few invariants and then storing the bytes as-is. I think reverting to `from_bytes` could be more clear. WDYT? (Sorry I missed this in #554!)